### ISSUE TITLE
s2e Utils.py

### ISSUE THREAD

**Issue by kxk-4498** [Demfier/multimodal-speech-emotion-recognition#19]
in utils.py of s2e it says test if test or train, but we don't have files such as moditifed_df_train/test.csv when doing pre-processing. Also when we try to run lstm-classifier(s2e) by changing name of modified_df to modified_df_train/test, we get the problem in float.tensor(input batch) of str in utils. 
 I dropped wav file, unnamed: 0 and label from df to make it work but nowit says i"nput.size(-1) must be equal to input_size. Expected 8, got 7 ".

**Comment by Demfier** [comment]
Hi @kxk-4498 

As for the first query, you'll have to create the test and train CSV files by first downloading the IEMOCAP dataset from the original authors and then following the notebooks (until `4_prepare_data.ipynb`) to create the dataset. Unfortunately, the IEMOCAP license doesn't allow me to publically share those processed data files.

And I am not very sure of the second issue you are facing; if you don't have a CSV file, how can you run the `lstm_classifier.py`?

**Comment by kxk-4498** [comment]
First problem:-
see, i have the IMEOCAP dataset and i did run the 4 files of preprocessing till 4_prepare_data.ipynb but the problem is that the csv file created by 4_prepare_data.ipynb is in the name of modified_df.csv. The problem is that in utils.py :-

ftype = 'test' if test else 'train'
    df = pd.read_csv('{}modified_df_{}.csv'.format(file_dir, ftype))

these lines in utils.py are calling modified_df_train/test.csv which didn't form during the preprocessing. Have you  missed uploading any preprocessing code which would've divided the modified_df.csv into modified_df_train.csv and modified_df_test.csv. This is my first problem.


Second problem:-
 i'm running lstm_classifier.py by making 2 copies of modified_df.csv from preprocessed folder and saving both copies into s2e as modified_df_train.csv and modified_df_test.csv coz i didn't have a better solution as to how much to train and test(I'm a noob and i'm learning).
So, when i run the s2e lstm_classifier.py i encounter the error of :-

batches.append([torch.FloatTensor(input_batch),

ValueError: too many dimensions 'str'

i removed this problem by dropping wav_file column from df in utils.py
then i got the error:

input.size(-1) must be equal to input_size. Expected 8, got 9

Then i removed label column as well
then i got the error:

The size of tensor a (6) must match the size of tensor b (1567) at non-singleton dimension 1

So, basically my second problem is due to the first problem that i don't have modified_df_train.csv and modified_df_test.csv in my s2e folder because it was not created during preprocessing and i have no idea how to fix it.

could you please give me your email-id so that i can send screenshots of the problems encountered.

the files i have in my folders are:
preprocessed:
audio_features.csv
audio_vectors_1.pkl
audio_vectors_2.pkl
2. audio_vectors_1.pkl
2. audio_vectors_1.pkl

**Comment by kxk-4498** [comment]
the files i have in my folders are:
preprocessed:
audio_features.csv
audio_vectors_1.pkl
audio_vectors_2.pkl
the files i have in my folders are:
preprocessed:
audio_features.csv
audio_vectors_1.pkl
audio_vectors_2.pkl

**Comment by kxk-4498** [comment]
First problem:-
see, i have the IMEOCAP dataset and i did run the 4 files of preprocessing till 4_prepare_data.ipynb but the problem is that the csv file created by 4_prepare_data.ipynb is in the name of modified_df.csv. The problem is that in utils.py :-

ftype = 'test' if test else 'train'
df = pd.read_csv('{}modified_df_{}.csv'.format(file_dir, ftype))

these lines in utils.py are calling modified_df_train/test.csv which didn't form during the preprocessing. Have you missed uploading any preprocessing code which would've divided the modified_df.csv into modified_df_train.csv and modified_df_test.csv. This is my first problem.

Second problem:-
i'm running lstm_classifier.py by making 2 copies of modified_df.csv from preprocessed folder and saving both copies into s2e as modified_df_train.csv and modified_df_test.csv coz i didn't have a better solution as to how much to train and test(I'm a noob and i'm learning).
So, when i run the s2e lstm_classifier.py i encounter the error of :-

batches.append([torch.FloatTensor(input_batch),

ValueError: too many dimensions 'str'

i removed this problem by dropping wav_file column from df in utils.py
then i got the error:

input.size(-1) must be equal to input_size. Expected 8, got 9

Then i removed label column as well
then i got the error:

The size of tensor a (6) must match the size of tensor b (1567) at non-singleton dimension 1

So, basically my second problem is due to the first problem that i don't have modified_df_train.csv and modified_df_test.csv in my s2e folder because it was not created during preprocessing and i have no idea how to fix it.

could you please give me your email-id so that i can send screenshots of the problems encountered.

the files i have in my folders are:
preprocessed:
audio_features.csv
audio_vectors_1.pkl
audio_vectors_2.pkl
audio_vectors_3.pkl
audio_vectors_4.pkl
audio_vectors_5.pkl
df_iemocap.csv
modified_df.csv
no_sample_df.csv

s2e:
audio_test.csv
audio_train.csv

t2e:
audiocode2text.pkl
text_test.csv
text_train.csv

pred_probas:
lr_classifier.pkl
mlp_classifier.pkl
mnb_classifier.pkl
rf_classifier.pkl
svc_classifier_model.pkl
xgb_classifier.pkl
text_lr_classifier.pkl
text_mlp_classifier.pkl
text_mnb_classifier.pkl
text_rf_classifier.pkl
text_svc_classifier_model.pkl
text_xgb_classifier.pkl
combined_features.pkl
combined_lr_classifier.pkl
combined_mlp_classifier.pkl
combined_mnb_classifier.pkl
combined_rf_classifier.pkl
combined_svc_classifier_model.pkl
combined_xgb_classifier.pkl

i changed the paths for some as per my convinience. Please help me run the lstm classifiier and please send me your email_id so that i can send you screenshots of the problems faced. thank you

**Comment by kaaaaate009** [comment]
> First problem:-
> see, i have the IMEOCAP dataset and i did run the 4 files of preprocessing till 4_prepare_data.ipynb but the problem is that the csv file created by 4_prepare_data.ipynb is in the name of modified_df.csv. The problem is that in utils.py :-
> 
> ftype = 'test' if test else 'train'
> df = pd.read_csv('{}modified_df_{}.csv'.format(file_dir, ftype))
> 
> these lines in utils.py are calling modified_df_train/test.csv which didn't form during the preprocessing. Have you missed uploading any preprocessing code which would've divided the modified_df.csv into modified_df_train.csv and modified_df_test.csv. This is my first problem.
> 
> Second problem:-
> i'm running lstm_classifier.py by making 2 copies of modified_df.csv from preprocessed folder and saving both copies into s2e as modified_df_train.csv and modified_df_test.csv coz i didn't have a better solution as to how much to train and test(I'm a noob and i'm learning).
> So, when i run the s2e lstm_classifier.py i encounter the error of :-
> 
> batches.append([torch.FloatTensor(input_batch),
> 
> ValueError: too many dimensions 'str'
> 
> i removed this problem by dropping wav_file column from df in utils.py
> then i got the error:
> 
> input.size(-1) must be equal to input_size. Expected 8, got 9
> 
> Then i removed label column as well
> then i got the error:
> 
> The size of tensor a (6) must match the size of tensor b (1567) at non-singleton dimension 1
> 
> So, basically my second problem is due to the first problem that i don't have modified_df_train.csv and modified_df_test.csv in my s2e folder because it was not created during preprocessing and i have no idea how to fix it.
> 
> could you please give me your email-id so that i can send screenshots of the problems encountered.

Actually had the same issue, cannot produce the modified_df_train/test.csv files, further leading to similar issues, even after renaming it.

**Comment by Demfier** [comment]
Hi @kaaaaate009,

The two files `modified_df_train.csv` and `modified_df_test.csv` were generated by splitting `modified_df.csv` obtained from script `4_prepare_data_.ipynb` I think I had used an 80:20 split.

**Comment by wineone** [comment]
hello, I think I solved the problem by changing the name of the csv's like above and line 17 of the utils.py file:
`data = (np.array(df[df.columns[1:]]), np.array(df[df.columns[0]]))`
for this:
`data = (np.array(df[df.columns[2:]]), np.array(df[df.columns[1]]))`
i think that was an error of index of features columns and sentiment column (output).
now it's running, i'm waiting for see if new error will be appear.

**Comment by Demfier** [comment]
You'll also need to remove the Unnamed columns from the csv files or else it won't work. I am closing this issue. Feel free to reopen it if the same error occurs again.
